packages/catalog-process/src/services/catalogService.ts

```diff
@@ -470,17 +470,17 @@ export function catalogServiceBuilder(
       const updatedTechnology = apiTechnologyToTechnology(
         eserviceSeed.technology
       );
-      if (eservice.data.descriptors.length === 1) {
-        const draftDescriptor = eservice.data.descriptors[0];
-        if (
-          updatedTechnology !== eservice.data.technology &&
-          draftDescriptor.interface !== undefined
-        ) {
-          await fileManager.delete(
-            config.s3Bucket,
-            draftDescriptor.interface.path
-          );
-        }
+      const interfaceHasToBeDeleted =
+        updatedTechnology !== eservice.data.technology &&
+        eservice.data.descriptors.length === 1 &&
+        eservice.data.descriptors[0].interface !== undefined;
+
+      if (interfaceHasToBeDeleted) {
+        await fileManager.delete(
+          config.s3Bucket,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          eservice.data.descriptors[0].interface!.path
+        );
       }
```

Comment:
just a minor suggestion, we could use the [same approach](https://github.com/pagopa/interop-be-monorepo/pull/292/files?diff=split&w=1#diff-77a5bad1f3b877e7bfc968a9b423c16649fca358d54286d743fe34fa555564ceR500) here
```suggestion
        updatedTechnology !== eservice.data.technology;

      if (interfaceHasToBeDeleted) {
        await Promise.all(eservice.data.descriptors.map(
          async d => {
            if (d.interface != undefined)
              return await fileManager.delete(config.s3Bucket, d.interface.path)
          }
        ))
      }
```

note: 
- this is possible only because we [already asserted](https://github.com/pagopa/interop-be-monorepo/pull/292/files?diff=split&w=1#diff-77a5bad1f3b877e7bfc968a9b423c16649fca358d54286d743fe34fa555564ceR457) that the eservice is in draft
- the `Promise.all` is a bit overkill in this case (but necessary)

Reply:
Updated